llvm/lib/Target/RISCV/RISCVVLOptimizer.cpp

```diff
@@ -732,6 +747,30 @@ static bool isSupportedInstr(const MachineInstr &MI) {
     return false;
 
   switch (RVV->BaseInstr) {
+  // Vector Unit-Stride Instructions
+  // Vector Strided Instructions
+  case RISCV::VLE8_V:
+  case RISCV::VLM_V:
+  case RISCV::VLSE8_V:
+  case RISCV::VLE8FF_V:
```

Comment:
Is it safe to reduce the VL of a fault-only-first load? I think that would potentially affect the result of its output VL operand.

Reply:
> These instructions execute as a regular load except that they will only take a trap caused by a synchronous exception on element 0. If element 0 raises an exception, vl is not modified, and the trap is taken. If an element > 0 raises an exception, the corresponding trap is not taken, and the vector length vl is reduced to the index of the element that would have raised an exception.

My thought was that it should not affect the result. Is this an incorrect understanding?

Reply:
It will affect the resulting VL though. I'm thinking if someone wrote something like:

```llvm
  %a = call { <vscale x 1 x i64>, i64 } @llvm.riscv.vleff.nxv1i64(<vscale x 1 x i64> undef, ptr %0, i64 %avl)
  %vl.ff = extractvalue { <vscale x 1 x i64>, i64 } %a, 1
```

%vl.ff might change if the vleff had its AVL reduced.

Reply:
If it had its AVL reduced, which caused a fault which would have happened to no longer occur, would this produce a different result?

Reply:
Yeah exactly

Reply:
Got it. Will remove.

Reply:
We can't reduce the VL of a load if the load was marked volatile in IR. We need to check the volatile flag of the memory operand.

Reply:
added the check and a test